Upstream merge

nwmac opened the pull request:
Merge latest code from upstream

nwmac committed:
Fix permission issue in docker all in one image (#4770)

richard-cox committed:
Update website version, tweak github action (#4771)

nwmac committed:
Improve the presentation of advanced options in endpoint registration (#4766)

nwmac committed:
Reduce size of text and icons in page header (#4758)

nwmac committed:
Use a subtle color for the app page header (#4765)

* Reduce size of text and icons in page header

* Change app page header color

nwmac committed:
Add support for no authentication (#4386)

* Add supprot for no authentication

* Fix backend unit tests

* Fix backend test failure

* Address PR feedback

* Address PR feedback

* Fix backend unit test mock

* Turn off auto-logout if the user can not log off

* Ensure local auth user can edit profile

* Fix unit tests

Co-authored-by: Richard Cox <richard.cox@suse.com>

richard-cox committed:
Upgrade to Angular 10 (#4724)

* update packages pre angular update

* Fix TS 4.x error

* After ng update @angular/core @angular/cli

* update @angular/cdk for ng update @angular/material

* after ng update @angular/material

* @angular-builders/custom-webpack fix
- ngtypecheck.ts part of build not used
- import `webpack` not known

* Row fix part 1

* ngrx-store-localstorage upgrade

* after ng update @ngrx/store

* after ng update @swimlane/ngx-charts

* After @swimlane/ngx-graph bump

* fixes for npm install warnings

* row fixes part 2

* Fix most of  common js deps, bring in typings
- seems the build now complains when bringing in js depedencies
  ```
  CommonJS or AMD dependencies can cause optimization bailouts.
  For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies
  ```
- the fix is to allow them, though we should look to moving away int eh future
- dependencies are...
  - moment-timezone.... us
  - js-yaml.... us
  - webcola & dagre... ngx-graph
  - all others come from @cfstratos/ajsf-core

* Ensure any cards that override abtract properties with @Input have their get & set

* Lint fixes

* commas not semi colons in object types

* Roll back TableCellCustomComponent naming change change

* changes following self review

* Roll back ngx-monaco-editor from 10.0.0-beta.1 to 9.0.0
- beta 1 is missing loader.js file, which is loaded by beta 1
- see https://github.com/atularen/ngx-monaco-editor/issues/185#issuecomment-690393639
- also tidy location of monaco assets

* Fix unit tests

* Fix Application Deploy (problem #2)

* Add debig for Application Deploy (problem #1)

* Fix Org Spaces List

* Fix Space Routes List

* Safer api keys test

* fix app wall test

* Fix Application Wall Tests

* More debug for Application Deploy

* Add more debugging

* Revert "Add more debugging"

This reverts commit 41bd7a0042a0293a17db68559b15d3becec58446.

* Revert "More debug for Application Deploy"

This reverts commit 008e91ed6a832a94fedc5c17ac114dbc28347bd4.

* Revert "Add debig for Application Deploy (problem #1)"

This reverts commit 14e53c06ce3b2283149ecc61c2d2eec01828366b.

nwmac committed:
Improve home page view (#4740)

* Allow k8s namespaces to be added as favourites

* Improve the home page

* Fix lint issue

* Tweaks and refactoring

* Add separator to layout menu

* Improve incremental loading

* Various tidyups and improvements

* Tidy ups and improvemts to async loading

* Fix progress spinner alignment

* Unscubribe

* Minor bug fixes and a few tidy ups

* Further tidy ups

* Fix front end unit tests

* Unit test fix

* Further unit tests fixes

* Endpoint card unit test fixes

* Fix unit tests

* Tidy up favorite card

* Further refinement to the UI

* Fix missing dates from recents on CF view

* Get app deploy from home screen working

* Add message for when no connected endpoints

* Unit test fixes

* Kubernetes Home Card unit test fixes

* Fix unit test error

* Move components to remove dependency on large shared module

* Fix unit test and add deploy tiles to CF Home Card

* Fix word wrap on favorite name

* Add tool tip

* Style tidy ups

* Minor tidy ups

* Fix test imports

* Fix unit tests

* e2e Debugging

* Fix for org delete e2e test

* E2e Test fix

* Address PR feedback so far

* Address PR feedback

* Fix connected endpoint state check on home screen

* Address PR feedback

* Restore apps link

* Fix unit test

nwmac committed:
Prevent password show/hide icon from receiving focus (#4776)

* Prevent password show/hide icon from receiving focus

* Fix lint issues

richard-cox committed:
Create defaults list component (#3987)

* Port downstream list filter changes
- https://github.com/SUSE/stratos/pull/185
- Apply to feature flag table

* Fix e2e tests

* Create defaults list component
- Intermediate component between simple list and full list
- Supply as little config as simple list, all of full list or anywhere in between

* Fix test code

* Major WIP

* Simple change

* Trivial change

* Trivial Change

* WIP

* WIP

* Tidying up

* Avoid chicken & egg linting issue

* Fix unit tests

* Fix another test

* Fix bug where all users were being fetched instead of just those in a cf
- best case both cf's connected as admins and call takes a long time
- worst case one cf isn't admin and call fails

* Fix pick --> omit

* Fix typing

* Fixes after merge

* WIP

* Fix unit test

* Fixes after merge

* Add comments, tidy up

* Fix unit tests

dependabot-preview[bot] committed:
Bump gopkg.in/DATA-DOG/go-sqlmock.v1 in /src/jetstream (#4733)

Bumps [gopkg.in/DATA-DOG/go-sqlmock.v1](https://github.com/DATA-DOG/go-sqlmock) from 1.3.0 to 1.5.0.
- [Release notes](https://github.com/DATA-DOG/go-sqlmock/releases)
- [Commits](https://github.com/DATA-DOG/go-sqlmock/compare/v1.3.0...v1.5.0)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

Co-authored-by: dependabot-preview[bot] <27856297+dependabot-preview[bot]@users.noreply.github.com>

dependabot-preview[bot] committed:
Bump github.com/golang/mock from 1.2.0 to 1.4.4 in /src/jetstream (#4732)

Bumps [github.com/golang/mock](https://github.com/golang/mock) from 1.2.0 to 1.4.4.
- [Release notes](https://github.com/golang/mock/releases)
- [Changelog](https://github.com/golang/mock/blob/master/.goreleaser.yml)
- [Commits](https://github.com/golang/mock/compare/v1.2.0...v1.4.4)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

Co-authored-by: dependabot-preview[bot] <27856297+dependabot-preview[bot]@users.noreply.github.com>

nwmac committed:
Fix lint issue (#4783)

ikapelyukhin committed:
Endpoint plugins configurable via YAML (#4751)

* Add proxy single request API endpoint

* Add endpoint plugin generation from a YAML file

* Deploy to CF from private GitHub and GitLab repos

* Use subtypes for YAML generated endpoints

* Nicer YAML format; ignore unknown endpoint types in the DB

nwmac committed:
Debug issues with mc CLI and minio server (#4790)

richard-cox committed:
Move git tools and some components into their own package (#4752)

* Move git tools and some components into their own package
- this then becomes more of a git library
- includes git
  - entities (and new git endpoint currently hidden), effects, actions, etc
  - commit list base items
  - git scm service and related items

* fix unit tests

* Update public_api.ts

* Update public_api.ts

* Fix lint issues

Co-authored-by: Neil MacDougall <neil.macdougall@suse.com>
Co-authored-by: Neil MacDougall <nwmac@users.noreply.github.com>

nwmac committed:
Don't fail if we can't access the GitHub API (#4779)

nwmac committed:
Fix ngDestroy bug on list view component (#4781)

nwmac committed:
Fix size and position of header notification badge icon (#4778)

* Fix size and position of header notification badge icon

* Ensure override also applies to dark mode
- dark mode brings back in i think specific overrides by material design
- ensure our overrides are more specific
- flicking between light/dark mode now shows no difference

Co-authored-by: Richard Cox <richard.cox@suse.com>

nwmac committed:
Fix helm endpoint link (#4777)

* Prevent password show/hide icon from receiving focus

* Fix lint issues

* Clicking on a helm repository endpoint should fitler to show it's charts only

* Fix for unit test

* Address PR feedback

nwmac committed:
Tidy up and refactor user favorites (#4757)

* Allow k8s namespaces to be added as favourites

* Improve the home page

* Fix lint issue

* Tweaks and refactoring

* Add separator to layout menu

* Improve incremental loading

* Various tidyups and improvements

* Tidy ups and improvemts to async loading

* Fix progress spinner alignment

* Unscubribe

* Minor bug fixes and a few tidy ups

* Further tidy ups

* Fix front end unit tests

* Unit test fix

* Further unit tests fixes

* Endpoint card unit test fixes

* Fix unit tests

* Tidy up favorite card

* Further refinement to the UI

* Fix missing dates from recents on CF view

* Get app deploy from home screen working

* Add message for when no connected endpoints

* Unit test fixes

* Kubernetes Home Card unit test fixes

* Fix unit test error

* Move components to remove dependency on large shared module

* Fix unit test and add deploy tiles to CF Home Card

* Fix word wrap on favorite name

* Add tool tip

* Style tidy ups

* Minor tidy ups

* Check favorites exists before navigating to them

* Fix test imports

* First round of tidy ups of the user favorites code

* Further refactor to reduce the complexity of user favorites

* Fix unit tests

* Allow Kubernetes namespaces to be favorited

* Fix duplicate import

* Tidy ups

* e2e Debugging

* Fix for org delete e2e test

* E2e Test fix

* Refactor out common entity operator

* Slight refactor

* Fix frontend unit tests

* Address PR feedback so far

* Address PR feedback

* Fix connected endpoint state check on home screen

* Address PR feedback

* Restore apps link

* Improve typing

* Fix undefined error

* Fix unit test

* Fix ordering issue

* Add a default home card

* Fix frong end lint issues

* Merge fixes

* Fix unit tests

* Fix lint issue

* Fix lint issue

* Fix unit test

* Fix merge issue

* Fix merge issue

* Fix kube favorites not showing and sizing issue

* Fix kube endpoint card shortcuts

* Reinstate entity validation checks for CF app/org/space

* Fix favorite validation

* Tidy ups

* Fix merge issue

* Remove unrelated change

* Address PR feedback

* Bug fixes

* Address PR feedback

* Fix link for kubernetes endpoints

nwmac committed:
Merge remote-tracking branch 'upstream/master'

nwmac committed:
Move test docker app off of docker hub to avoid rate limiting (#4802)

* Move test docker app off of docker hub to avoid rate limiting

* Fix image reference

richard-cox committed:
Handle table cell get/setters better (#4799)

- get and set were moved into abstract base class as part of ts bump
- we now need to ensure we get/set the correct way
- also contains minor cell tidy ups

nwmac committed:
Fix link for metrics endpoints (#4800)

nwmac committed:
Fix issue where username/password are not encoded for basic auth (#4796)

nwmac committed:
Metrics: Ensure Stratos Metrics metadata file is optional (#4801)

* Fix link for metrics endpoints

* Ensure metrics works with plain Prometheus

* Remove test code

* Slight optimization in if

nwmac committed:
Update terminology for SSO Accept List and add to Helm Chart docs (#4797)

* Docuement SSO Accept List and rename terminology

* Use Allow not Accept

* Update log message

* Remove debug logging

* Document SSO_OPTIONS

nwmac committed:
Fix double build when using ng serve (#4805)

nwmac committed:
Fix alignment of no content mesasges on home page view (#4809)

tealeg committed:
fix unicode rendering in application logstreams (#4808)

nwmac committed:
User Favorites: Further refactoring to simplify the code base (#4798)

* Allow k8s namespaces to be added as favourites

* Improve the home page

* Fix lint issue

* Tweaks and refactoring

* Add separator to layout menu

* Improve incremental loading

* Various tidyups and improvements

* Tidy ups and improvemts to async loading

* Fix progress spinner alignment

* Unscubribe

* Minor bug fixes and a few tidy ups

* Further tidy ups

* Fix front end unit tests

* Unit test fix

* Further unit tests fixes

* Endpoint card unit test fixes

* Fix unit tests

* Tidy up favorite card

* Further refinement to the UI

* Fix missing dates from recents on CF view

* Get app deploy from home screen working

* Add message for when no connected endpoints

* Unit test fixes

* Kubernetes Home Card unit test fixes

* Fix unit test error

* Move components to remove dependency on large shared module

* Fix unit test and add deploy tiles to CF Home Card

* Fix word wrap on favorite name

* Add tool tip

* Style tidy ups

* Minor tidy ups

* Check favorites exists before navigating to them

* Fix test imports

* First round of tidy ups of the user favorites code

* Further refactor to reduce the complexity of user favorites

* Fix unit tests

* Allow Kubernetes namespaces to be favorited

* Fix duplicate import

* Tidy ups

* e2e Debugging

* Fix for org delete e2e test

* E2e Test fix

* Refactor out common entity operator

* Slight refactor

* Fix frontend unit tests

* Address PR feedback so far

* Address PR feedback

* Fix connected endpoint state check on home screen

* Address PR feedback

* Restore apps link

* Improve typing

* Fix undefined error

* Fix unit test

* Fix ordering issue

* Add a default home card

* Fix frong end lint issues

* Merge fixes

* Fix unit tests

* Fix lint issue

* Fix lint issue

* Fix unit test

* Fix merge issue

* Fix merge issue

* Fix kube favorites not showing and sizing issue

* Fix kube endpoint card shortcuts

* Reinstate entity validation checks for CF app/org/space

* Fix favorite validation

* Tidy ups

* Fix merge issue

* Remove unrelated change

* Address PR feedback

* Bug fixes

* User Favorites: More refactoring to simplify

* Fix merge issues

* More merge fixes

* Fix issue with front-end unit tests

* Fix unit test issue

* Remove unused import

PG2000 committed:
chore: remove trailing comma as this is a non standard (#4815)

* remove trailing comma as this is a non standard

* remove trailing comma as this is a non standard

* remove trailing comma as this is a non standard

nwmac committed:
Allow backup/restore plugin to be enabled/disabled (#4818)

nwmac committed:
Home page: Make configurable showing all endpoints or just favorites (#4810)

* Fix alignment of no content mesasges on home page view

* Allow home view to show all endpoints are just favorites

* Fix unit tests

* FIx backend test

* Changes following review

* Fix backend unit test

Co-authored-by: Richard Cox <richard.cox@suse.com>

nwmac committed:
Deploy Applications from Enterprise GitHub and GitLab (#4782)

* Frontend for new yaml based git endpoints
- register github.com and gitlab.com endpoints and connect with credentials
- register and connect to github and gitlan enterprise instances
- select these as sources when deploying a cf application

* Add proxy single request API endpoint

* Add endpoint plugin generation from a YAML file

* Deploy to CF from private GitHub and GitLab repos

* Use subtypes for YAML generated endpoints

* Nicer YAML format; ignore unknown endpoint types in the DB

* Fix merge issues and add icon indicator for private repositories

* Fix lint issues

* Add details component to show username and avatar on endpoint card

* Various fixes and improvements

* Fix lint issue

* Fix merge issues

* Unit test fixes

* Unit test fix following merge

* One more unit test fix

* Fix backend unit tests

* Fix the deploy types todo (in a verbose way)

* Ignore commit id when possible

* Fix redeploy stepper

* Fix minor bugs in new register git stepper

* multiple fixes, remaining todos

* update github/gitlab connect readme's, including scope info

* In connect helper remove duped header

* Fix multiple gitlab issues

* Fix deploy app home screen link for github/gitlab with credentials

* Ensure app git tab only shows if we have access to repo
- can happen if others view app deployed via private repo

* Fix clone url & improve error messaging

* Improve cf app summary page git info

* Fix issue where...
- we're waiting for an endpoint that doesn't exist.. in an effect
- that endpoint then is created
- the effect then fires off the request... long after it should have stopped

* Fix multiple gitlab enterprise issues
- enterprise gits weren't showing as deploy source types
- enterprise gits that weren't connected failed to work (fetching data & deploying)

* Fix git enterprise ssl setting, provide hint at time of url required

* Show private repo indicator on app git tab

* app git tab: handle git fetch failure beter
- also rework observables in page

* Changes following quick self review

* Fix linting

* Fixes following testing

* Use enum for github string

* Ensure Git entities associated with an endpoint are removed on endpoint unregister/disconnect

* Fix unit test

* Tidy up backend changes

* Update app summary page

* Show no content message if app git tab cannot show info

* display password field instead of text area for token

* other changes following review

* Fix other issues raised in review (all from single comment)

* Fix frontend after updated backend

* Apply path escaping fix when url is recreated

* Fix some of the e2e tests

* Fixes after testing

* Fix more e2e tests

* Changes following review

Co-authored-by: Richard Cox <richard.cox@suse.com>
Co-authored-by: Ivan Kapelyukhin <ikapelyukhin@suse.com>

dependabot-preview[bot] committed:
Bump gopkg.in/yaml.v2 from 2.3.0 to 2.4.0 in /src/jetstream (#4817)

Bumps [gopkg.in/yaml.v2](https://github.com/go-yaml/yaml) from 2.3.0 to 2.4.0.
- [Release notes](https://github.com/go-yaml/yaml/releases)
- [Commits](https://github.com/go-yaml/yaml/compare/v2.3.0...v2.4.0)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

Co-authored-by: dependabot-preview[bot] <27856297+dependabot-preview[bot]@users.noreply.github.com>

nwmac committed:
Fix ever-increasing array of snackbar references (#4820)

* Fix ever-increasing array of snackbar references

* Fix isse where latest is always null

* Fix snackbar service

nwmac committed:
K8s add support for more resources (#4780)

* k8s: Add support for more resource types in a generic way

* Unit test fix

* WIP

* Fixes for pods view

* Tidy ups

* Add more resource types

* Fix lint issues

* Fix build issue

* Fix unit test

* Unit test fix

* Fix unit tests

* Fix lint issue

* RE-arrange kube entity creation
- now assign directly to kube entity catalog
- to access custom kube generator resources have to do some funnies in
  kubernetes-tab-base.component.ts

* Fix services in namespace view
- kube guid and namespace was mixed up
- now follow standard pattern of kube guid then namespace in ctor

* Changes following review

* Fix unit tests

Co-authored-by: Richard Cox <richard.cox@suse.com>

richard-cox committed:
Persist list settings over refresh (#4405)

* WIP

* Add list reset button & list button tooltips

* Few tidy ups, add clear local storage feature

* Tweak profile settings section

* WIP

* WIP

* only store pagination sections from lists

* Fix clear and auto select multi filter/s on reset

* Ingore invalid values

* Super basic encryption of pag section

* fix reset

* Fix max list count & multi-entity selectors stickyness

* tidying up

* Fix linting

* Fix unit tests

* Changes following review #1

* Changes following review #2

* Changes following review no. 3

* Fix warning on first load

* Fix show org/space e2e failures

* improve delete service e2e tests

* Ensure list is card mode before waiting to find a card

* Remove stray line

richard-cox committed:
Merge remote-tracking branch 'upstream/master' into upstream-merge

richard-cox reviewed (approved):
Found one specific downstream thing we might want to fix
- helm page: the filter panel on the left highlights which repo is selected, this is a green amongst blue links. could be improved

Discovered some bugs, but they're upstream issues
- helm chart appscode / kubedb doesn't load, should be investigated. Should also show failure on page, currently get forever spinner
- helm chart list - resetting filters/sort fails to remove the selected repo
- home screen kube card - clicking on pods metric card gives a 404

richard-cox committed:
Fix linting

richard-cox committed:
Don't use generic url validation for kube endpoints (#4826)

- this was recently added
- false negative for minikube url (https://192.168.39.159:8443)

richard-cox committed:
Improvement and fix for no-content component (#4827)

* Improvements and fixes for no-content component
- add colour to improve monotone views on first load
- fix issue where no-content header arrow failed to show in production
  (see https://github.com/angular/angular-cli/issues/17440)

* Fix another opacity percentage
- seen in import kube table edit endpoint name cell

dependabot-preview[bot] committed:
Bump github.com/mattn/go-sqlite3 from 1.13.0 to 1.14.5 in /src/jetstream (#4816)

Bumps [github.com/mattn/go-sqlite3](https://github.com/mattn/go-sqlite3) from 1.13.0 to 1.14.5.
- [Release notes](https://github.com/mattn/go-sqlite3/releases)
- [Commits](https://github.com/mattn/go-sqlite3/compare/v1.13.0...v1.14.5)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

Co-authored-by: dependabot-preview[bot] <27856297+dependabot-preview[bot]@users.noreply.github.com>

ikapelyukhin committed:
Document deployment with an ingress; make ingress host optional (#4793)

richard-cox committed:
Minor Kube Fixes (#4829)

* Fix home screen kube card pod link

* Ensure reset list filter button also resets helm chart repo filter

* Ensure we handle the case where the versions info is unable

* Fix link for namespaces

Co-authored-by: Neil MacDougall <nwmac@users.noreply.github.com>

nwmac committed:
K8s: More resource support - refinements (#4822)

* Allow k8s namespaces to be added as favourites

* k8s: Add support for more resource types in a generic way

* Improve the home page

* Fix lint issue

* Tweaks and refactoring

* Add separator to layout menu

* Improve incremental loading

* Various tidyups and improvements

* Tidy ups and improvemts to async loading

* Fix progress spinner alignment

* Unscubribe

* Minor bug fixes and a few tidy ups

* Further tidy ups

* Fix front end unit tests

* Unit test fix

* Further unit tests fixes

* Endpoint card unit test fixes

* Fix unit tests

* Tidy up favorite card

* Further refinement to the UI

* Fix missing dates from recents on CF view

* Get app deploy from home screen working

* Add message for when no connected endpoints

* Unit test fixes

* Kubernetes Home Card unit test fixes

* Fix unit test error

* Move components to remove dependency on large shared module

* Fix unit test and add deploy tiles to CF Home Card

* Fix word wrap on favorite name

* Add tool tip

* Style tidy ups

* Minor tidy ups

* Check favorites exists before navigating to them

* Fix test imports

* First round of tidy ups of the user favorites code

* Further refactor to reduce the complexity of user favorites

* Fix unit tests

* Allow Kubernetes namespaces to be favorited

* Fix duplicate import

* Tidy ups

* e2e Debugging

* Fix for org delete e2e test

* E2e Test fix

* Refactor out common entity operator

* Slight refactor

* Fix frontend unit tests

* Unit test fix

* Address PR feedback so far

* Address PR feedback

* Fix connected endpoint state check on home screen

* Address PR feedback

* Restore apps link

* Improve typing

* Fix undefined error

* Fix unit test

* Fix ordering issue

* Add a default home card

* Fix frong end lint issues

* Merge fixes

* Fix unit tests

* WIP

* Fixes for pods view

* Tidy ups

* Add more resource types

* Fix lint issue

* Fix lint issue

* Fix unit test

* Fix lint issues

* Fix build issue

* Fix unit test

* Unit test fix

* Fix unit tests

* More improvements

* Numerous improvements

* Fix lint issue

* RE-arrange kube entity creation
- now assign directly to kube entity catalog
- to access custom kube generator resources have to do some funnies in
  kubernetes-tab-base.component.ts

* Fix services in namespace view
- kube guid and namespace was mixed up
- now follow standard pattern of kube guid then namespace in ctor

* Fix merge issues

* Changes following review

* Fix unit tests

* Fix merge issue

* Fix merge issues

* Changes following merge

* Changes following merge

* Fix merge conflicts

* Only show analysis link when enabled

* Fix issue favoriting from side panel

* Fix expression changed error

* Fix expression changed errors

* Fix a few minor issues and unit test

* Remove old namespace components

* Remvoe TODO

* Delete fixes
- fixes to delete action param
- added to delete pods
- successEntityHandler doesn't run, so have to manually clear pagination sections

* Fix delete ns
- ensure delete action has correct guid
- ensure response has kube guid

* Fix expanding pod row

* Fix show of analysis in namespace resource viewer

* Fix fav namespace link

* Fix PR issues

Co-authored-by: Richard Cox <richard.cox@suse.com>

richard-cox committed:
Add additional information to new kube resource tables (#4825)

* Allow k8s namespaces to be added as favourites

* k8s: Add support for more resource types in a generic way

* Improve the home page

* Fix lint issue

* Tweaks and refactoring

* Add separator to layout menu

* Improve incremental loading

* Various tidyups and improvements

* Tidy ups and improvemts to async loading

* Fix progress spinner alignment

* Unscubribe

* Minor bug fixes and a few tidy ups

* Further tidy ups

* Fix front end unit tests

* Unit test fix

* Further unit tests fixes

* Endpoint card unit test fixes

* Fix unit tests

* Tidy up favorite card

* Further refinement to the UI

* Fix missing dates from recents on CF view

* Get app deploy from home screen working

* Add message for when no connected endpoints

* Unit test fixes

* Kubernetes Home Card unit test fixes

* Fix unit test error

* Move components to remove dependency on large shared module

* Fix unit test and add deploy tiles to CF Home Card

* Fix word wrap on favorite name

* Add tool tip

* Style tidy ups

* Minor tidy ups

* Check favorites exists before navigating to them

* Fix test imports

* First round of tidy ups of the user favorites code

* Further refactor to reduce the complexity of user favorites

* Fix unit tests

* Allow Kubernetes namespaces to be favorited

* Fix duplicate import

* Tidy ups

* e2e Debugging

* Fix for org delete e2e test

* E2e Test fix

* Refactor out common entity operator

* Slight refactor

* Fix frontend unit tests

* Unit test fix

* Address PR feedback so far

* Address PR feedback

* Fix connected endpoint state check on home screen

* Address PR feedback

* Restore apps link

* Improve typing

* Fix undefined error

* Fix unit test

* Fix ordering issue

* Add a default home card

* Fix frong end lint issues

* Merge fixes

* Fix unit tests

* WIP

* Fixes for pods view

* Tidy ups

* Add more resource types

* Fix lint issue

* Fix lint issue

* Fix unit test

* Fix lint issues

* Fix build issue

* Fix unit test

* Unit test fix

* Fix unit tests

* More improvements

* Numerous improvements

* Fix lint issue

* RE-arrange kube entity creation
- now assign directly to kube entity catalog
- to access custom kube generator resources have to do some funnies in
  kubernetes-tab-base.component.ts

* Fix services in namespace view
- kube guid and namespace was mixed up
- now follow standard pattern of kube guid then namespace in ctor

* Fix merge issues

* Changes following review

* Fix unit tests

* Fix merge issue

* Fix merge issues

* Changes following merge

* Changes following merge

* Add additional information to new kube resource tables

* Fix merge conflicts

* Only show analysis link when enabled

* Fix issue favoriting from side panel

* Fix expression changed error

* Fix expression changed errors

* Fix a few minor issues and unit test

* Remove old namespace components

* Remvoe TODO

* Fix merge issue

* Fix PR issues

* Delete fixes
- fixes to delete action param
- added to delete pods
- successEntityHandler doesn't run, so have to manually clear pagination sections

* Fix delete ns
- ensure delete action has correct guid
- ensure response has kube guid

* Don't use generic url validation for kube endpoints (#4826)

- this was recently added
- false negative for minikube url (https://192.168.39.159:8443)

* Improvement and fix for no-content component (#4827)

* Improvements and fixes for no-content component
- add colour to improve monotone views on first load
- fix issue where no-content header arrow failed to show in production
  (see https://github.com/angular/angular-cli/issues/17440)

* Fix another opacity percentage
- seen in import kube table edit endpoint name cell

* Bump github.com/mattn/go-sqlite3 from 1.13.0 to 1.14.5 in /src/jetstream (#4816)

Bumps [github.com/mattn/go-sqlite3](https://github.com/mattn/go-sqlite3) from 1.13.0 to 1.14.5.
- [Release notes](https://github.com/mattn/go-sqlite3/releases)
- [Commits](https://github.com/mattn/go-sqlite3/compare/v1.13.0...v1.14.5)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

Co-authored-by: dependabot-preview[bot] <27856297+dependabot-preview[bot]@users.noreply.github.com>

* Document deployment with an ingress; make ingress host optional (#4793)

* Minor Kube Fixes (#4829)

* Fix home screen kube card pod link

* Ensure reset list filter button also resets helm chart repo filter

* Ensure we handle the case where the versions info is unable

* Fix link for namespaces

Co-authored-by: Neil MacDougall <nwmac@users.noreply.github.com>

* Fix expanding pod row

* Fix show of analysis in namespace resource viewer

* Fix fav namespace link

Co-authored-by: Neil MacDougall <neil.macdougall@suse.com>
Co-authored-by: Neil MacDougall <neil_macdougall@hotmail.com>
Co-authored-by: dependabot-preview[bot] <27856297+dependabot-preview[bot]@users.noreply.github.com>
Co-authored-by: Ivan Kapelyukhin <ikapelyukhin@suse.com>
Co-authored-by: Neil MacDougall <nwmac@users.noreply.github.com>

bgandon committed:
Improve SSO documentation with detailed UAA setup (#4833)

Document how to properly setup UAA client in production environment, with
proper listing in login home page.

richard-cox committed:
Add additional kube resources to Workload page (#4830)

* Allow k8s namespaces to be added as favourites

* k8s: Add support for more resource types in a generic way

* Improve the home page

* Fix lint issue

* Tweaks and refactoring

* Add separator to layout menu

* Improve incremental loading

* Various tidyups and improvements

* Tidy ups and improvemts to async loading

* Fix progress spinner alignment

* Unscubribe

* Minor bug fixes and a few tidy ups

* Further tidy ups

* Fix front end unit tests

* Unit test fix

* Further unit tests fixes

* Endpoint card unit test fixes

* Fix unit tests

* Tidy up favorite card

* Further refinement to the UI

* Fix missing dates from recents on CF view

* Get app deploy from home screen working

* Add message for when no connected endpoints

* Unit test fixes

* Kubernetes Home Card unit test fixes

* Fix unit test error

* Move components to remove dependency on large shared module

* Fix unit test and add deploy tiles to CF Home Card

* Fix word wrap on favorite name

* Add tool tip

* Style tidy ups

* Minor tidy ups

* Check favorites exists before navigating to them

* Fix test imports

* First round of tidy ups of the user favorites code

* Further refactor to reduce the complexity of user favorites

* Fix unit tests

* Allow Kubernetes namespaces to be favorited

* Fix duplicate import

* Tidy ups

* e2e Debugging

* Fix for org delete e2e test

* E2e Test fix

* Refactor out common entity operator

* Slight refactor

* Fix frontend unit tests

* Unit test fix

* Address PR feedback so far

* Address PR feedback

* Fix connected endpoint state check on home screen

* Address PR feedback

* Restore apps link

* Improve typing

* Fix undefined error

* Fix unit test

* Fix ordering issue

* Add a default home card

* Fix frong end lint issues

* Merge fixes

* Fix unit tests

* WIP

* Fixes for pods view

* Tidy ups

* Add more resource types

* Fix lint issue

* Fix lint issue

* Fix unit test

* Fix lint issues

* Fix build issue

* Fix unit test

* Unit test fix

* Fix unit tests

* More improvements

* Numerous improvements

* Fix lint issue

* RE-arrange kube entity creation
- now assign directly to kube entity catalog
- to access custom kube generator resources have to do some funnies in
  kubernetes-tab-base.component.ts

* Fix services in namespace view
- kube guid and namespace was mixed up
- now follow standard pattern of kube guid then namespace in ctor

* Fix merge issues

* Changes following review

* Fix unit tests

* Fix merge issue

* Fix merge issues

* Changes following merge

* Changes following merge

* Fix merge conflicts

* Only show analysis link when enabled

* Fix issue favoriting from side panel

* Fix expression changed error

* Fix expression changed errors

* Fix a few minor issues and unit test

* Remove old namespace components

* Remvoe TODO

* Add additional kube resources to Workload page
Two small todos left
- tidy up process in helm-release-socket-service
- switch service and pods view over to new process and remove old components/files

* Convert workload pods and services view to standard resource views

* Tidy up HelmReleaseSocketService

* Delete fixes
- fixes to delete action param
- added to delete pods
- successEntityHandler doesn't run, so have to manually clear pagination sections

* Fix delete ns
- ensure delete action has correct guid
- ensure response has kube guid

* Fix expanding pod row

* Fix show of analysis in namespace resource viewer

* Fix fav namespace link

* Fix PR issues

* remove fluff from kube generator

* Fixes post merge

* Add additional information to new kube resource tables (#4825)

* Allow k8s namespaces to be added as favourites

* k8s: Add support for more resource types in a generic way

* Improve the home page

* Fix lint issue

* Tweaks and refactoring

* Add separator to layout menu

* Improve incremental loading

* Various tidyups and improvements

* Tidy ups and improvemts to async loading

* Fix progress spinner alignment

* Unscubribe

* Minor bug fixes and a few tidy ups

* Further tidy ups

* Fix front end unit tests

* Unit test fix

* Further unit tests fixes

* Endpoint card unit test fixes

* Fix unit tests

* Tidy up favorite card

* Further refinement to the UI

* Fix missing dates from recents on CF view

* Get app deploy from home screen working

* Add message for when no connected endpoints

* Unit test fixes

* Kubernetes Home Card unit test fixes

* Fix unit test error

* Move components to remove dependency on large shared module

* Fix unit test and add deploy tiles to CF Home Card

* Fix word wrap on favorite name

* Add tool tip

* Style tidy ups

* Minor tidy ups

* Check favorites exists before navigating to them

* Fix test imports

* First round of tidy ups of the user favorites code

* Further refactor to reduce the complexity of user favorites

* Fix unit tests

* Allow Kubernetes namespaces to be favorited

* Fix duplicate import

* Tidy ups

* e2e Debugging

* Fix for org delete e2e test

* E2e Test fix

* Refactor out common entity operator

* Slight refactor

* Fix frontend unit tests

* Unit test fix

* Address PR feedback so far

* Address PR feedback

* Fix connected endpoint state check on home screen

* Address PR feedback

* Restore apps link

* Improve typing

* Fix undefined error

* Fix unit test

* Fix ordering issue

* Add a default home card

* Fix frong end lint issues

* Merge fixes

* Fix unit tests

* WIP

* Fixes for pods view

* Tidy ups

* Add more resource types

* Fix lint issue

* Fix lint issue

* Fix unit test

* Fix lint issues

* Fix build issue

* Fix unit test

* Unit test fix

* Fix unit tests

* More improvements

* Numerous improvements

* Fix lint issue

* RE-arrange kube entity creation
- now assign directly to kube entity catalog
- to access custom kube generator resources have to do some funnies in
  kubernetes-tab-base.component.ts

* Fix services in namespace view
- kube guid and namespace was mixed up
- now follow standard pattern of kube guid then namespace in ctor

* Fix merge issues

* Changes following review

* Fix unit tests

* Fix merge issue

* Fix merge issues

* Changes following merge

* Changes following merge

* Add additional information to new kube resource tables

* Fix merge conflicts

* Only show analysis link when enabled

* Fix issue favoriting from side panel

* Fix expression changed error

* Fix expression changed errors

* Fix a few minor issues and unit test

* Remove old namespace components

* Remvoe TODO

* Fix merge issue

* Fix PR issues

* Delete fixes
- fixes to delete action param
- added to delete pods
- successEntityHandler doesn't run, so have to manually clear pagination sections

* Fix delete ns
- ensure delete action has correct guid
- ensure response has kube guid

* Don't use generic url validation for kube endpoints (#4826)

- this was recently added
- false negative for minikube url (https://192.168.39.159:8443)

* Improvement and fix for no-content component (#4827)

* Improvements and fixes for no-content component
- add colour to improve monotone views on first load
- fix issue where no-content header arrow failed to show in production
  (see https://github.com/angular/angular-cli/issues/17440)

* Fix another opacity percentage
- seen in import kube table edit endpoint name cell

* Bump github.com/mattn/go-sqlite3 from 1.13.0 to 1.14.5 in /src/jetstream (#4816)

Bumps [github.com/mattn/go-sqlite3](https://github.com/mattn/go-sqlite3) from 1.13.0 to 1.14.5.
- [Release notes](https://github.com/mattn/go-sqlite3/releases)
- [Commits](https://github.com/mattn/go-sqlite3/compare/v1.13.0...v1.14.5)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

Co-authored-by: dependabot-preview[bot] <27856297+dependabot-preview[bot]@users.noreply.github.com>

* Document deployment with an ingress; make ingress host optional (#4793)

* Minor Kube Fixes (#4829)

* Fix home screen kube card pod link

* Ensure reset list filter button also resets helm chart repo filter

* Ensure we handle the case where the versions info is unable

* Fix link for namespaces

Co-authored-by: Neil MacDougall <nwmac@users.noreply.github.com>

* Fix expanding pod row

* Fix show of analysis in namespace resource viewer

* Fix fav namespace link

Co-authored-by: Neil MacDougall <neil.macdougall@suse.com>
Co-authored-by: Neil MacDougall <neil_macdougall@hotmail.com>
Co-authored-by: dependabot-preview[bot] <27856297+dependabot-preview[bot]@users.noreply.github.com>
Co-authored-by: Ivan Kapelyukhin <ikapelyukhin@suse.com>
Co-authored-by: Neil MacDougall <nwmac@users.noreply.github.com>

* Add comment after review

Co-authored-by: Neil MacDougall <neil.macdougall@suse.com>
Co-authored-by: Neil MacDougall <neil_macdougall@hotmail.com>
Co-authored-by: dependabot-preview[bot] <27856297+dependabot-preview[bot]@users.noreply.github.com>
Co-authored-by: Ivan Kapelyukhin <ikapelyukhin@suse.com>
Co-authored-by: Neil MacDougall <nwmac@users.noreply.github.com>

nwmac committed:
Electron update (#4811)

* Add supprot for no authentication

* Electron WIP

* Electron app WIP

* Add packaging

* Wait for backend to start

* Non-functional tidy ups

* Move desktop theme into desktop-extentions
- see run.sh for some enable/disable shinanigans

* Disable auto-log out when there's no auth
- need to tidy up how 'no auth' is determined

* Add Local flag

* Remove jetstream binary

* Allow SessionExpiry to be customised

* Add SESSION_STORE_EXPIRY customisation

* Re-enable loggin service
- should now be a no-op for no auth scenario (session never expires)

* Logging change

* Disable edit of local endpoints
- all the infomation is dynamically created

* Add local endpoint indication

* Allow packages to only supply routing module

* Split out profile settings into it's own component

* Add back in dark mode

* Add desktop settings page accessible via menu

* Fix SESSION_STORE_EXPIRY

* Add icon

* Add correct icon

* Add blank login page

* Add basic snackbar integration
- Send OS notification instead of snackbar (only covers snackbar service, not all snackbar ref uses)
- brings in ngx-electron to get easy isElectron test and access to ipcRenderer
- ignores 'return' type used to navigate to area of app

* Add a 'install' nav item into desktop-extensions
- this can be split out into a separate package

* Tweaks

* Tweak endpoint card

* Remove unused loadsh

* A fwe tweaks. Better about page

* Add file watch

* Add auto-update support

* Bug fixes for cf

* Only listen for endpointsChanged when running with eletron

* Mirror CF Disconnect/Unregister in config

* Add warning when disconnecting/unregistering local endpoints

* Improve look of DMG

* Tweak alert badge. Remove copyright icon

* Remove info logging that was debug

* Tweaks

* Persist last location, start from location

* Fix path

* Minor tweaks

* Tweaks

* Fix lint issues

* Fix compilation issues

* Fix backend test compilation

* Fix front-end unit tests

* Add test coverage file

* Fix build issue

* Fix merge issue

* Fix backend unit tests

* Fix setting bug and read version from file

* Exclude desktop extension by default

* More improvements

* Improvements for initial checkout

* Tweak to clean checkout experience

* Tidy up

* Bump version number to 4.4.0

* Fix bug that broke id on user menu button

* Fixes following merge

* Fix deploy app snack bar

* Ensure backend plugin is included

* Add kube desktop support

* Fix unit tests

* Fix backend build issue

* A few minor fixes

* Fix issue reading version file

Co-authored-by: Richard Cox <richard.cox@suse.com>

dependabot[bot] committed:
Bump ini from 1.3.5 to 1.3.7 in /electron (#4837)

Bumps [ini](https://github.com/isaacs/ini) from 1.3.5 to 1.3.7.
- [Release notes](https://github.com/isaacs/ini/releases)
- [Commits](https://github.com/isaacs/ini/compare/v1.3.5...v1.3.7)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

richard-cox committed:
Update versions and change log (#4835)

richard-cox committed:
Fix edit profile button visibility (#4839)

- broken as part of electron changes
- canEdit$ was incorrectly removed

richard-cox committed:
Merge remote-tracking branch 'upstream/master' into upstream-merge

richard-cox committed:
Ensure initial/default plugins.yaml is included in build (#4840)

nwmac committed:
Ensure we copy plugins.yaml for AIO and Cloud Foundry (#4841)

richard-cox committed:
Merge remote-tracking branch 'upstream/master' into upstream-merge
